sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java

```diff
@@ -0,0 +1,190 @@
+    private final int LPS_THS_P_H = 0x31; //  Pressure interrupt threshold, high byte
+
+    // The next two registers need special soldering
+    private final int LPS_RPDS_L = 0x39;//  Pressure offset for differential pressure computing, low byte
+    private final int LPS_RPDS_H = 0x3A; //  Differential offset, high byte
+    private int mMillibarAdjust = 0;
+
+    /**
+     * Create a new barometric pressure and temperature sensor driver connected on the given I2C bus.
+     *
+     * @param bus I2C bus the sensor is connected to.
+     * @throws IOException when a lower level does
+     */
+    public BaroTemp(String bus) throws IOException {
+        PeripheralManager pioService = PeripheralManager.getInstance();
+        mDevice = pioService.openI2cDevice(bus, SenseHat.I2C_LPS25H_ADDRESS);
+        // power down first
+        mDevice.writeRegByte(LPS_CTRL_REG1, (byte) 0);
+        // power up, data rate 12.5Hz (10110000)
+        mDevice.writeRegByte(LPS_CTRL_REG1, (byte) 0xb0);
+        if (0xBD != (mDevice.readRegByte(LPS_WHO_AM_I) & 0xFF)) {
+            throw new IOException("This does not seem to be a LPS25H");
+        }
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (mDevice != null) {
+            try {
+                // power down device
+                mDevice.writeRegByte(LPS_CTRL_REG1, (byte) 0);
+            } catch (Exception any) {
+                // we tried
+            }
+            try {
+                mDevice.close();
+            } finally {
+                mDevice = null;
+            }
+        }
+    }
+
+    private int readSigned24(int a0, int a1, int a2) throws IOException {
+        int ret = (mDevice.readRegByte(a0) & 0xFF);
+        ret += ((int) mDevice.readRegByte(a1) & 0xFF) << 8;
+        ret += ((int) mDevice.readRegByte(a2) & 0xFF) << 16;
+        if (ret < 8388608) return ret;
+        else return ret - 16777216;
+    }
+
+    private int readSigned16(int a0, int a1) throws IOException {
+        int ret = (mDevice.readRegByte(a0) & 0xFF);
+        ret += (mDevice.readRegByte(a1) & 0xFF) << 8;
+        if (ret < 32768) return ret;
+        else return ret - 65536;
+    }
+
+    /**
+     * The sensor seems to have an offset to the actual pressure. You can find your local "real"
+     * pressure quite easily on the web. Get the measured value from the sensor and compute the
+     * difference. The value obtained can be passed to this method to "calibrate" your board's
+     * sensor. In the author's case the difference was 6.2 hPa which is quite significant. This
+     * error was confirmed with another (unrelated) sensor.
+     *
+     * @param hPa difference to actual air pressure in hectoPascal (hPa) or millibar.
+     * @throws IOException from I2cDevice
+     */
+    public void setBarometerOffset(double hPa) throws IOException {
+        this.mMillibarAdjust = (int) Math.round(hPa * 4096);
```

Comment:
The use of `this` is not necessary as the `m` prefix to the variable name implies it is part of the class.